app/src/auth.py

```diff
@@ -28,7 +28,14 @@ def login_post():
     teacher = Teacher.query.filter_by(email=email).first()
 
     # Determine if the email belongs to a Teacher or a User
-    account = teacher if teacher else user
+    if teacher and not user:
+        account = teacher
+    elif user and not teacher:
+        account = user
+    else:
```

Comment:
Change the else to `elif user and teacher`; otherwise, if a user tries to log in with an invalid email, the proper login error prompt at line 51 will never be raised and this "multiple roles" one will be shown instead.